### ISSUE TITLE
Problem 1.3 release and dynamically generating a class file

### ISSUE THREAD

**Issue by larrymelia** [julianthome/inmemantlr#10]
I'm getting this dump:

 -------------------
Exception in thread "main" java.lang.IllegalArgumentException: bytecode of class GenScalaFacadesFromTypescriptLexer is empty
	at org.snt.inmemantlr.comp.SpecialJavaFileManager.getByteCodeFromClass(SpecialJavaFileManager.java:99)
	at org.snt.inmemantlr.comp.StringCompiler.compile(StringCompiler.java:135)
	at org.snt.inmemantlr.GenericParser.compile(GenericParser.java:293)
	at com.cyberthinkers.tools.GenScala$.parseSource(GenScalaFacadesFromTypescriptBaseListenerScala.scala:36)
	at com.cyberthinkers.tools.GenScala$.main(GenScalaFacadesFromTypescriptBaseListenerScala.scala:27)
	at com.cyberthinkers.tools.GenScala.main(GenScalaFacadesFromTypescriptBaseListenerScala.scala)
 -------------------
I have a TypeScript ambient grammar that I wrote, and your library is crashing when your librar tries to compile it. It works correctly when I use the standard library.

**Comment by julianthome** [comment]
Hello Larry,

would it be possible for you to share your g4 file so that I can reproduce and fix the bug?

Thank you very much and kind regards,
  Julian

**Comment by larrymelia** [comment]
File attached.

**Comment by julianthome** [comment]
Hello Larry,

sorry, I could not find an attached file. Could you sent it via E-mail to me (julian.thome.de@gmail.com).

Thanks a lot for your help and kind regards,
  Julian

**Comment by larrymelia** [comment]
Here's the Github link for my project and g4 file:
https://github.com/cyberthinkers/typescript-to-scala-js-facade/tree/master/src/main/antlr4

**Comment by julianthome** [comment]
Hello Larry,

thanks for the link. I've just checked the grammar with inmemantlr-v1.3.1 and with vanilla ANTLR v4.6. In both cases, I do get the same error, though.

```bash
cd /tmp
sudo curl -O http://www.antlr.org/download/antlr-4.6-complete.jar
export CLASSPATH=".:/tmp/antlr-4.6-complete.jar:$CLASSPATH"
alias antlr4='java -jar /tmp/antlr-4.6-complete.jar'
antlr4 GenScalaFacadesFromTypescript.g4
javac -classpath '/tmp/antlr-4.6-complete.jar:.' *.java
```

The response is:
```
GenScalaFacadesFromTypescriptLexer.java:156: error: cannot find symbol
			return !strictMode;
			        ^
  symbol:   variable strictMode
  location: class GenScalaFacadesFromTypescriptLexer
1 error
```

After removing the `{!strictMode}` reference in the g4 file, it worked for both inmemantlr and vanilla ANTLR.

I was wondering if you could provide me some information on how you've got the grammar working with vanilla ANTLR. 


Thank you very much and kind regards,
  Julian

**Comment by larrymelia** [comment]
Okay, this is a bug in their generator that was supposed to be fixed but
seems to be still an issue. I've been manually fixing it after generating
the parser. Let me check to see if there's another update, and if not I'll
post a bug on their site.

**Comment by larrymelia** [comment]
It's working now after I updated the ANTLR lib version to 4.6.

**Comment by julianthome** [comment]
Ah I see. I think the [ECMA script grammar](https://github.com/antlr/grammars-v4/tree/master/ecmascript) from the grammar-v4 package might be a useful resource as well since they also use the strict member of the Lexer class.

The grammars-v4 cases are handled by the JUnit class [TestExternalGrammars](https://github.com/julianthome/inmemantlr/blob/master/inmemantlr-api/src/test/java/TestExternalGrammars.java) (the ECMA script test case can be found in the function `testEcmaScript`).

Kind regards,
  Julian

**Comment by larrymelia** [comment]
Hi Julian,

The bug I mentioned in ANTLR 4 was fixed but never pushed to Marven
Central. The earlier version I required a fix every time I generated a new
Java parser from the grammar. Anyway, as you mentioned, the "strict" flag
handles some kind of special lexer processing, which is not documented
anywhere. Thanks for your help and your library. It's a life saver I hope!
I'll probably need to have some email exchanges in the future.

Larry
